### PR TITLE
Allowed font_style to be an array

### LINKED ISSUE #49: Using multiple font styles
Is there a way to use multiple `font-style`s at once, for example `bold` together with `underline`?

### PR DESCRIPTION
**Pull request by John-Toohey**
This allows multiple `font_style`s to be used, allowing, say, for a bold and italic text. Fixes #49

All previous examples should work, but now you can also specify:

```yaml
font-style:
  - bold
  - underline
```

### CODE REVIEW

File: src/theme.rs [hunk 1]
```diff
@@ -123,13 +102,10 @@ impl Theme {
 
             let background = transpose(background.map(|fg| self.get_color(fg)))?;
 
-            let mut style: String = format!("{font_style}", font_style = *font_style_ansi,);
+            let mut style: String = format!("{font_style}");
```
Comment (sharkdp) [inline src/theme.rs:105]: This is not quite what we had before, right? If there is no foreground color and no background color, there will now be a trailing semicolon.

Comment (sharkdp) [reply src/theme.rs:105]: Ah, I believe the tests are also failing because of this.

Comment (John-Toohey) [reply src/theme.rs:105]: I believe I have fixed this.

File: src/font_style.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+impl Display for FontStyle {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        for (i, style) in self.0.iter().enumerate() {
+            if i + 1 == self.0.len() {
+                write!(f, "{}", style)?;
+            } else {
+                write!(f, "{};", style)?;
+            }
+        }
```
Comment (sharkdp) [inline src/font_style.rs:66]: Would it be possible to use https://doc.rust-lang.org/std/primitive.slice.html#method.join here?

Comment (John-Toohey) [reply src/font_style.rs:66]: Nope, because that would only work if the slice was a slice of `str`s.

Comment (sharkdp) [reply src/font_style.rs:66]: ah, right. we could use intersperse from itertools, but it's not worth a new dependency. Maybe once it is stabilized in `std` :smile: (https://github.com/rust-lang/rust/issues/79524, https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.intersperse)

**Comment by sharkdp** [comment]
Thank you for the update

**Pull request merged**